velox/functions/sparksql/Hash.cpp

```diff
@@ -378,10 +378,33 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> hashSignatures() {
               .build()};
 }
 
+void checkInputType(const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  for (int i = 0; i < inputArgs.size(); i++) {
+    switch (inputArgs[i].type->kind()) {
+      case TypeKind::BOOLEAN:
```

Comment:
Is this a limitation in Spark as well?

Let's update documentation to clarify.

Reply:
Hi @mbasmanova, the limitations actually come from the implementation for hash function in this file. For example, complex data types are not really supported here. We would like to let `makeHash` fail if unsupported input type is used.

Reply:
All of production use cases use complex types. Consider adding support without any further delay.

Reply:
I see. We will spare some time to add the support in separate PRs. Thanks!